bench/workgen/workgen.cxx

```diff
@@ -902,9 +919,30 @@ int ThreadRunner::op_run(Operation *op) {
     if (op->_transaction != NULL) {
         if (ret != 0 || op->_transaction->_rollback)
             WT_TRET(_session->rollback_transaction(_session, NULL));
-        else if (_in_transaction)
-            ret = _session->commit_transaction(_session,
-              op->_transaction->_commit_config.c_str());
+        else if (_in_transaction) {
+            // Set prepare, commit and durable timestamp if prepare is set.
+            if(op->_transaction->use_prepare) {
+                workgen_epoch(&start_time);
+                _time_us = ts_us(start_time);
+                sprintf(time_buf, "prepare_timestamp=%" PRIu64, _time_us);
+                ret = _session->prepare_transaction(_session, time_buf);
+                sleep(1);
```

Comment:
The sleep of a second feels quite heavy-weight to simulate an application that may do thousands of transactions a second.  Let's make this use `usleep()` (microseconds), and also give the test writer control over the time spent.  To do the latter, let's change `use_prepare` to be a double.  That will be the number of seconds, and by making it a double, the caller can specify small delays.  I think you could take `ceil(use_prepare * 1000000.0)` and treat that as the number of microseconds.  If it's set to 0.0 (the default), there is no prepare done.  Maybe change the name of `use_prepare` to `prepare`, since the former sounds more like a boolean.  It could also be `prepare_time` if that seems clearer to you.

Reply:
Fixed. Updated the code to take `prepare_time` in `secs(double)` and converting into `microseconds`, using `usleep`.